tensorflow/python/framework/ops.py

```diff
@@ -4737,6 +4737,10 @@ def my_func(pred, tensor):
     """
     if control_inputs is None:
       return self._ControlDependenciesController(self, None)
+
+    # Imported here to avoid circular dependency.
+    from tensorflow.python.ops import resource_variable_ops  # pylint: disable=g-import-not-at-top
+
```

Comment:
For the `AutoCastVariable` use case only the func_graph changes are strictly necessary. I only included these changes for consistency, but I am happy to revert which would fix any circular dependencies.